cabal-install/bootstrap.sh

```diff
@@ -181,8 +181,18 @@ PARSEC_VER="3.1.9";    PARSEC_VER_REGEXP="[3]\.[01]\."
                        # >= 3.0 && < 3.2
 DEEPSEQ_VER="1.4.2.0"; DEEPSEQ_VER_REGEXP="1\.[1-9]\."
                        # >= 1.1 && < 2
-BYTESTRING_VER="0.10.8.1"; BYTESTRING_VER_REGEXP="0\.10\.[2-9]"
-                       # >=0.10.2
+
+case "$GHC_VER" in
```

Comment:
I think we can remove all mentions of `bytestring` from the bootstrap script if we add a similar conditional for `binary` instead. So GHC <= 7.6 will use `binary-0.8.2.1` (which depends on `bytestring >= 0.9` and therefore should work with the [versions of `bytestring` that come with GHC 7.6 and 7.4](https://ghc.haskell.org/trac/ghc/wiki/Commentary/Libraries/VersionHistory)) and GHC > 7.6 will use `binary-0.8.3.0`.